### DISCUSSION TITLE
Callback to Variable

### DISCUSSION THREAD

**Discussion by AndreyKl4** [FreeOpcUa/python-opcua#1221] [Q&A]
Hi all!

I have faced such problem: 
I have made an opcua server and added some variables. After that i connected to this server with UaExpert Client. I changed some values of variables in Client and i want to monitor changes in Server side. The question is: in what way i can observe in server side, what variables were changed? Is in python-opcua any callback function to do this? 
To solve this problem i found 2 solutions:
1) To observe all variables constantly and find out differences...
2) To write the method and call it after every change of variables in Client side... 
But I think it has to be a more clever solution:)

Thanks in advance!

Andrii

**Comment by oroulet** [comment]
you can subscribe on server side. so you get a callback for every variable change

**Reply by AndreyKl4** [reply]
@oroulet Thanks for the quick answer! For subscription i have to use https://github.com/FreeOpcUa/python-opcua/blob/master/examples/client-example.py this example. Am i right? sorry for stupid questions. I am really new with opcua...

**Reply by AndreasHeine** [reply]
yes and no you need a serverside subscription!

https://github.com/FreeOpcUa/python-opcua/blob/1cca33dd5fb70858b5e811b08356ca97520bcbaf/opcua/server/server.py#L388

so in difference to the client example you posted you need to call create_subscription on the server class/obj the rest is the same like in the client example

**Reply by AndreyKl4** [reply]
@AndreasHeine thanks a lot for your answer!
